version: '3.8'

services:
  app:
    build:
      context: .

    ports:
      - 8000:8000
    command: pipenv run python manage.py runserver 0.0.0.0:8000
    volumes:
      - .:/home/python/app
    depends_on:
      db:
        condition: service_healthy
  db:
    image: postgres:alpine
    volumes:
      - ./database:/var/lib/postgresql/data
    ports:
      - 5432:5432
    environment:
      - POSTGRES_USER=${POSTGRES_USER}
      - POSTGRES_PASSWORD=${POSTGRES_PASSWORD}
    env_file:
      - .env
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U ${POSTGRES_USER}"]
      interval: 3s
      timeout: 3s
      retries: 10